runtime/src/accounts_index.rs

```diff
@@ -160,72 +405,31 @@ impl<'a, T: 'a + Clone> AccountsIndex<T> {
         max_root
     }
 
-    pub fn insert(
-        &mut self,
-        slot: Slot,
-        pubkey: &Pubkey,
-        account_info: T,
-        reclaims: &mut SlotList<T>,
-    ) {
-        self.account_maps
-            .entry(*pubkey)
-            .or_insert_with(|| (AtomicU64::new(0), RwLock::new(SlotList::with_capacity(32))));
-        self.update(slot, pubkey, account_info, reclaims);
-    }
-
-    // Try to update an item in account_maps. If the account is not
-    // already present, then the function will return back Some(account_info) which
-    // the caller can then take the write lock and do an 'insert' with the item.
-    // It returns None if the item is already present and thus successfully updated.
-    pub fn update(
+    // Updates the given pubkey at the given slot with the new account information.
+    // Returns true if the pubkey was newly inserted into the index, otherwise, if the
+    // pubkey updates an existing entry in the index, returns false.
+    pub fn update_or_create_if_missing(
```

Comment:
nits: I welcome less confusing name so thanks for taking a bite on this. but this is a bit long relatively for frequently called method. how about just `upsert`?

Reply:
that works for me!